web/js/map-OpenLayers.js

```diff
@@ -115,6 +115,67 @@ var fixmystreet = fixmystreet || {};
         fixmystreet.markers.redraw();
       },
 
+      shortlist_multiple: function(ids, token) {
+        $.post("/my/planned/change_multiple", { ids: ids, token: token })
+        .done(function() {
+          var $itemList = $('.item-list'),
+              items = [];
+          $('.shortlisted-status').remove();
+
+          for (var i = 0; i < ids.length; i++) {
+            var problemId = ids[i],
+                $item = $itemList.find('#report-'+ problemId)
+                $form = $item.find('form'),
+                $submit = $form.find("input[type='submit']" );
+
+            $item.attr('data-lastupdate', new Date().toISOString());
+
+            $submit.attr('name', 'shortlist-remove')
+                   .attr('aria-label', $submit.data('label-remove'))
+                   .removeClass('item-list__item__shortlist-add')
+                   .addClass('item-list__item__shortlist-remove');
+            items.push({
+              'url': '/report/' + $item.data('report-id'),
+              'lastupdate': $item.data('lastupdate')
+            });
+          }
+          $(document).trigger('shortlist-all', { items: items});
+        })
+        .fail(function(response) {
+          if (response.status == 400) {
+            // If the response is 400, then get a new CSRF token and retry
+            var csrf = response.responseText.match(/name="token" value="([^"]*)"/)[1];
+            fixmystreet.maps.shortlist_multiple(ids, csrf);
```

Comment:
There is the possibility (if slight) of a loop here if the server were to keep returning 400 for some reason. Perhaps add a countdown flag so it won't retry for ever?